### PR TITLE
`get_mentions_state()` implemented on the Rust side (only for users and @room)

### PR DESCRIPTION
**Pull request by Velin92**
Implemented and exposed a function to get the current state of mentions in RTE.
This will allow us to efficiently get the current state of mentions, which might be very useful for intentional mentions.

### CODE REVIEW

File: crates/wysiwyg/src/composer_model/mentions.rs [hunk 1]
```diff
@@ -13,15 +13,43 @@
...
+        for node in self.state.dom.iter_mentions() {
+            match node.kind() {
+                MentionNodeKind::AtRoom => {
+                    mentions_state.has_at_room_mention = true
+                }
+                MentionNodeKind::MatrixUri { mention } => {
+                    match mention.kind() {
+                        matrix_mentions::MentionKind::Room => {
+                            // Not required yet
+                        }
```
Comment (jonnyandrew) [inline crates/wysiwyg/src/composer_model/mentions.rs:40]: The omission of rooms in the result could be a bit confusing given the name of the function because it sounds like it will return all mentions. Can I suggest either:
- Update the name of the function and result to `get_intentional_mentions() -> IntentionalMentions` and add documentation that references [MSC3952](https://github.com/matrix-org/matrix-spec-proposals/pull/3952) so that it's implied that only spec'ed mentions will be returned. Or,
- Include room IDs in the result and leave it up to the client to ignore them.

File: crates/wysiwyg/src/mentions_state.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
```
Comment (jonnyandrew) [inline crates/wysiwyg/src/mentions_state.rs:1]:
```suggestion
// Copyright 2023 The Matrix.org Foundation C.I.C.
```

File: crates/wysiwyg/src/tests/test_mentions.rs [hunk 3]
```diff
@@ -594,6 +594,73 @@ fn can_insert_at_room_mention() {
...
+fn get_mentions_state_for_user_mention() {
+    let model = cm("<p>hello <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a>!|</p>");
+    let mut state = MentionsState::default();
+    state.user_ids.insert("@alice:matrix.org".into());
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_multiple_user_mentions() {
+    let model = cm("<p>hello <a data-mention-type=\"user\" href=\"https://matrix.to/#/@alice:matrix.org\" contenteditable=\"false\">Alice</a> and <a data-mention-type=\"user\" href=\"https://matrix.to/#/@bob:matrix.org\" contenteditable=\"false\">Bob</a>!|</p>");
```
Comment (jonnyandrew) [inline crates/wysiwyg/src/tests/test_mentions.rs:613]: We shouldn't need to include `data-mention-type` and `contenteditable` in the input because the model should parse mentions automatically. Please can you try removing?

File: crates/wysiwyg/src/tests/test_mentions.rs [hunk 4]
```diff
@@ -594,6 +594,73 @@ fn can_insert_at_room_mention() {
...
+    let model = cm("<p>hello @room|");
+    let state = MentionsState {
+        user_ids: Default::default(),
+        has_at_room_mention: true,
+    };
+    assert_eq!(model.get_mentions_state(), state)
+}
+
+#[test]
+fn get_mentions_state_for_multiple_user_and_at_room_mentions() {
```
Comment (jonnyandrew) [inline crates/wysiwyg/src/tests/test_mentions.rs:641]: Should we add one test for content containing multiple `@room` mentions and duplicate user mentions?

**Review by jonnyandrew — APPROVED** [review]
Looks good to me! Just a few comments

File: crates/wysiwyg/src/composer_model/mentions.rs (hunk 1 above)
Comment (Velin92) [reply crates/wysiwyg/src/composer_model/mentions.rs:40]: Adding a way to also include `room_ids` and `room_aliases`
